src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -176,121 +176,129 @@ private void WriteGreenType(TreeType node)
+                            OverrideOrNewModifier(field), field.Type, field.Name, CamelCase(field.Name), i);
                     }
                     else if (field.Type == "SyntaxNodeOrTokenList")
                     {
-                        WriteLine("    public {0}Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode> {1} => new Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>(this.{2});",
-                            OverrideOrNewModifier(field), field.Name, CamelCase(field.Name)
-                            );
+                        WriteLine("public {0}Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode> {1} => new Microsoft.CodeAnalysis.Syntax.InternalSyntax.SyntaxList<CSharpSyntaxNode>(this.{2});",
+                            OverrideOrNewModifier(field), field.Name, CamelCase(field.Name));
                     }
                     else
                     {
-                        WriteLine("    public {0}{1} {2} => this.{3};",
-                            OverrideOrNewModifier(field), field.Type, field.Name, CamelCase(field.Name)
-                            );
+                        WriteLine("public {0}{1} {2} => this.{3};",
+                            OverrideOrNewModifier(field), field.Type, field.Name, CamelCase(field.Name));
                     }
                 }
 
                 for (int i = 0, n = valueFields.Count; i < n; i++)
                 {
                     var field = valueFields[i];
-                    WriteComment(field.PropertyComment, "    ");
-                    WriteLine("    public {0}{1} {2} => this.{3};",
-                        OverrideOrNewModifier(field), field.Type, field.Name, CamelCase(field.Name)
-                        );
+                    WriteComment(field.PropertyComment, "");
+                    WriteLine("public {0}{1} {2} => this.{3};",
+                        OverrideOrNewModifier(field), field.Type, field.Name, CamelCase(field.Name));
                 }
 
                 // GetSlot
                 WriteLine();
-                WriteLine("    internal override GreenNode GetSlot(int index)");
-                WriteLine("    {");
-                WriteLine("        switch (index)");
-                WriteLine("        {");
-                for (int i = 0, n = nodeFields.Count; i < n; i++)
+                Write("internal override GreenNode GetSlot(int index)");
+
+                if (nodeFields.Count == 0)
                 {
-                    var field = nodeFields[i];
-                    WriteLine("            case {0}: return this.{1};", i, CamelCase(field.Name));
+                    WriteLine(" => null;");
+                }
+                else if (nodeFields.Count == 1)
+                {
+                    WriteLine();
+                    Indent();
+                    WriteLine("=> index == 0 ? this.{0} : null;", CamelCase(nodeFields[0].Name));
```

Comment:
If we're adding new strings, I'd much prefer if we moved to interpolated strings for new things. Yes, it's inconsistent, but interpolated strings are much more readable for these applications.

Reply:
@333fred I would like ot follow this up with a PR that makes that change uniformly.  This is something i def intend to do :)

Reply:
Would you be amenable to me doing that in another PR just to keep the number of commits here under control?